core/test_network/src/lib.rs

```diff
@@ -436,8 +436,8 @@ where
         &mut self,
         configuration: Configuration,
         genesis: Option<G>,
-        instruction_validator: impl Into<IsInstructionAllowedBoxed<W>> + Send + 'static,
-        query_validator: impl Into<IsQueryAllowedBoxed<W>> + Send + 'static,
+        instruction_validator: impl Into<IsInstructionAllowedBoxed> + Send + 'static,
```

Comment:
By the way, I've just realized that the `WorldTrait` generic is not needed inside `Network` and `Peer` structures and can be hard-coded [here](https://github.com/hyperledger/iroha/blob/iroha2-dev/core/test_network/src/lib.rs#L462) because the tests use the default `World` structure.

Reply:
Yeah, now I removed `WorldTrait` generic from everywhere and it works fine